modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/maven/LightweightClassLoaderModelBuilder.java

```diff
@@ -57,17 +65,82 @@ public LightweightClassLoaderModelBuilder(File artifactFolder,
+    additionalDependencies.stream()
+        .map(BundleDependency::getDescriptor)
         .map(descriptor -> new org.mule.maven.client.api.model.BundleDescriptor.Builder()
             .setGroupId(descriptor.getGroupId())
             .setArtifactId(descriptor.getArtifactId())
             .setVersion(descriptor.getVersion())
             .setType(descriptor.getType())
             .setClassifier(descriptor.getClassifier().orElse(null)).build())
-        .map(bundleDescriptor -> mavenClient.resolveBundleDescriptor(bundleDescriptor))
-        .collect(toList());
+        .forEach(bundleDescriptor -> {
+          resolveDependency(bundleDescriptor)
+              .forEach(additionalDep -> updateAdditionalDependencyOrFail(resolvedAdditionalDependencies, additionalDep));
+        });
+    return resolvedAdditionalDependencies;
+  }
+
+  private List<org.mule.maven.client.api.model.BundleDependency> resolveDependency(org.mule.maven.client.api.model.BundleDescriptor dependencyBundleDescriptor) {
+    List<org.mule.maven.client.api.model.BundleDependency> resolvedDependencies = new ArrayList<>();
+    resolvedDependencies.add(mavenClient.resolveBundleDescriptor(dependencyBundleDescriptor));
+    resolvedDependencies.addAll(mavenClient.resolveBundleDescriptorDependencies(false, false, dependencyBundleDescriptor));
+    return resolvedDependencies;
+  }
+
+  private void updateAdditionalDependencyOrFail(List<org.mule.maven.client.api.model.BundleDependency> additionalDependencies,
+                                                org.mule.maven.client.api.model.BundleDependency bundleDependency) {
+    Reference<org.mule.maven.client.api.model.BundleDependency> replace = new Reference<>();
+    additionalDependencies.stream()
+        .filter(additionalBundleDependency -> StringUtils.equals(additionalBundleDependency.getDescriptor().getGroupId(),
+                                                                 bundleDependency.getDescriptor().getGroupId())
+            &&
+            StringUtils.equals(additionalBundleDependency.getDescriptor().getArtifactId(),
+                               bundleDependency.getDescriptor().getArtifactId()))
+        .findFirst()
+        .map(additionalBundleDependency -> {
+          String additionalBundleDependencyVersion = additionalBundleDependency.getDescriptor().getVersion();
+          String bundleDependencyVersion = bundleDependency.getDescriptor().getVersion();
+          if (areSameMajor(bundleDependencyVersion, additionalBundleDependencyVersion)) {
+            if (isNewerVersion(bundleDependencyVersion, additionalBundleDependencyVersion)) {
+              replace.set(additionalBundleDependency);
+            }
+          } else {
+            throw new MuleRuntimeException(createStaticMessage("Attempting to add different major versions of the same dependency as additional plugin dependency. If this is not explicitly defined, check transitive dependencies."
```

Comment:
There is no tests for this neither here or the related PR https://github.com/mulesoft/mule-ee-distributions/pull/547. Would it be possible to cover this scenario in an integration test?

Reply:
It's covered in the mule-maven-plugin PR, but i will add another tests in mule-ee-distributions